model-optimizer/extensions/front/onnx/loop_ext.py

```diff
@@ -57,38 +62,104 @@ def extract(cls, loop_node):
                     if inp == '':
                         # input is omitted; most likely it corresponds to an optional input for an operator
                         continue
-                    elif inp in main_graph.graph['tensor_mapping']:
-                        log.debug('The edge between outer and inner graphs detected: {} -> {}'.format(inp, id))
-                        if main_graph.graph['tensor_mapping'][inp] not in additional_params:
-                            # create new Parameter body node and connect the body node with the outer graph using it
-                            param_id = str(inp)
-                            body_graph.add_node(param_id, kind='op', op='Parameter', name=param_id, pb=None, shape=None)
-                            parameter_node = Node(body_graph, param_id)
-                            # need to manually update necessary attrs for the node because extractor will not be called
-                            # for it because the node does not have .pb attribute
-                            Parameter.update_node_stat(parameter_node, {})
-                            external_edges.append((main_graph.graph['tensor_mapping'][inp], parameter_node, inp))
-                            src_id, src_port = param_id, 0
-                            additional_params[main_graph.graph['tensor_mapping'][inp]] = parameter_node
-                        else:
-                            src_id, src_port = additional_params[main_graph.graph['tensor_mapping'][inp]].id, 0
                     else:
-                        raise Error('Reference to "{}" is not satisfied. A node refer not existing data tensor. ONNX '
-                                    'model is not consistent. Protobuf fragment: {}', inp, pb_node)
+                        cur_graph = body_graph
+                        counter = 0
+                        is_finished = False
+                        transit_parameter = None
+                        # go through all levels of nested graphs starting from the deepest
+                        while not is_finished and 'parent_node' in cur_graph.graph:
+                            parent_graph = cur_graph.graph['parent_node'].graph
+                            external_edges.append([])
+                            additional_params.append({})
+                            # if parent graph contains input node, create edge from outer to inner graph
+                            if inp in parent_graph.graph['tensor_mapping']:
+                                log.debug('The edge between outer and inner graphs detected: {} -> {}'.format(inp, id))
+                                # if parameter in inner graph already created, use it. Otherwise - create new one
```

Comment:
please add assert:
`assert 1 <= counter and counter  < additional_params.len() + 1`

Reply:
and below the same comment